airflow/api_connexion/openapi/v1.yaml

```diff
@@ -1572,6 +1572,192 @@ paths:
         '404':
           $ref: '#/components/responses/NotFound'
 
+  /auth-info:
+    get:
+      summary: Get site authentication information
+      x-openapi-router-controller: airflow.api_connexion.endpoints.auth_endpoint
+      operationId: get_auth_info
+      tags: [Authentication]
+      responses:
+        '200':
+          description: Success.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuthInfo'
+
+  /auth/login:
+    post:
+      summary: Login user
+      x-openapi-router-controller: airflow.api_connexion.endpoints.auth_endpoint
+      operationId: auth_login
+      tags: [Authentication]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AuthDBLoginForm'
+      responses:
+        '200':
+          description: Success.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JwtAuthAndUser'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+
+
+  /auth-oauth/{provider}:
+    get:
+      summary: OAUTH user login
+      x-openapi-router-controller: airflow.api_connexion.endpoints.auth_endpoint
+      operationId: auth_oauthlogin
+      tags: [Authentication]
+      parameters:
+        - $ref: '#/components/parameters/ShouldRegister'
+        - $ref: '#/components/parameters/OauthProvider'
+      responses:
+        '302':
+          description: Redirect.
+          headers:
+            Location:
+              schema:
+                type: string
+                format: uri
+
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+
+  /oauth-authorized:
+    get:
+      summary: Oauth authentication
+      x-openapi-router-controller: airflow.api_connexion.endpoints.auth_endpoint
+      operationId: authorize_oauth
+      tags: [Authentication]
+      responses:
+        '200':
+          description: Success.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JwtAuthAndUser'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
```

Comment:
Don't we need more status codes like `50X` which happens when the auth configs are wrongly setup & callback fails?